presto-orc/src/main/java/com/facebook/presto/orc/stream/LongInputStreamV1.java

```diff
@@ -130,6 +152,77 @@ public void skip(long items)
             long consume = Math.min(items, numLiterals - used);
             used += consume;
             items -= consume;
+            if (items != 0) {
+                // A skip of multiple runs can take place at seeking
+                // to checkpoint. Keep track of the start of the run
+                // in literals for use by next scan().
+                currentRunOffset += (int) consume;
+            }
+        }
+    }
+
+    public int scan(
+            int[] offsets,
+            int beginOffset,
+            int numOffsets,
+            int endOffset,
+            ResultsConsumer resultsConsumer)
+            throws IOException
+    {
+        this.offsets = offsets;
+        this.beginOffset = beginOffset;
+        this.numOffsets = numOffsets;
+        this.endOffset = endOffset;
+        this.resultsConsumer = resultsConsumer;
+        this.numResults = 0;
+        this.offsetIdx = beginOffset;
+
+        if (numLiterals > 0) {
+            scanLiterals();
+        }
+        while (offsetIdx < beginOffset + numOffsets) {
+            used = 0;
+            numLiterals = 0;
+            scanDone = false;
+            readValues();
+            if (!scanDone) {
```

Comment:
I don't see any code that sets `scanDone` to true. If it is always false, then it is not needed.